### PR TITLE
[CoreML EP] Add Resize Support

### PR DESCRIPTION
**Pull request by gwang-msft**
**Description**: [CoreML EP] Add Resize Support

**Motivation and Context**
- Resize is part of the 1p model (Office Lens)
- For now added only the Bilinear/NN upsample since these are the most popular resize usage so far
- Moved some util functions to shared space

### CODE REVIEW

File: onnxruntime/core/providers/coreml/builders/impl/resize_op_builder.cc [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+  if (!GetShape(*input_defs[0], input_shape, logger))
+    return false;
+
+  const auto input_size = input_shape.size();
+  if (input_size != 4) {
+    LOGS(logger, VERBOSE) << "Resize only support 4d shape, input is "
+                          << input_size << "d shape";
+    return false;
+  }
+
```
Comment (skottmckay) [inline onnxruntime/core/providers/coreml/builders/impl/resize_op_builder.cc:137]: When we create the mobile friendly model do we ever remove the batch dim so a 3d shape is possible? Or it just gets locked to being 1?

**Review by skottmckay — APPROVED** [review]
:shipit:

File: onnxruntime/core/providers/coreml/builders/impl/resize_op_builder.cc (hunk 1 above)
Comment (gwang-msft) [reply onnxruntime/core/providers/coreml/builders/impl/resize_op_builder.cc:137]: By convention, 2d image has 4d shape, 3d shape is of something like NCH which is 1d array/image

**Pull request merged**